include/TSPSolver.h

```diff
@@ -0,0 +1,20 @@
+#include "CostMatrix.h"
+
+struct TSPResult
+{
+    int cost;
+    std::vector<int> path;
+    TSPResult() = default;
+    TSPResult(std::vector<int> path, int cost) : path(path), cost(cost) {}
+};
+
+class TSPSolver
+{
+public:
+    TSPSolver() = default;
```

Comment:
declaring **default constructor** as default makes sense only if you declare some other constructor because it prevents compiler to generate this automatically.